website/docs/guides/ip_spaces.html.markdown

```diff
@@ -0,0 +1,286 @@
+    }
+  }
+
+  ip_range {
+    start_address = "11.11.11.100"
+    end_address   = "11.11.11.110"
+  }
+
+  ip_range {
+    start_address = "11.11.11.120"
+    end_address   = "11.11.11.123"
+  }
+}
+
+resource "vcd_external_network_v2" "provider-gateway" {
+  name = "T0-backed-provider-gateway"
+
+  nsxt_network {
+    nsxt_manager_id      = data.vcd_nsxt_manager.main.id
+    nsxt_tier0_router_id = data.vcd_nsxt_tier0_router.router.id
+  }
+
+  # boolean flag to enable IP Space support
+  use_ip_spaces = true
+}
+
+resource "vcd_ip_space_uplink" "u1" {
+  name                = "IP Space Uplink assignment"
+  external_network_id = vcd_external_network_v2.provider-gateway.id
+  ip_space_id         = vcd_ip_space.space1.id
+}
+
+resource "vcd_nsxt_edgegateway" "ip-space" {
+  org                 = "cloud"
+  name                = "ip-space-backed-edge"
+  owner_id            = data.vcd_org_vdc.vdc1.id
+  external_network_id = vcd_external_network_v2.provider-gateway.id
+
+  # Explicit dependency to be sure that IP Space Uplink is configured before
+  # configuring Edge Gateway
+  depends_on = [vcd_ip_space_uplink.u1]
+}
+
+resource "vcd_ip_space_ip_allocation" "public-floating-ip" {
+  org_id      = data.vcd_org.org1.id
+  ip_space_id = vcd_ip_space.space1.id
+  type        = "FLOATING_IP"
+
+  depends_on = [vcd_nsxt_edgegateway.ip-space]
+}
+
+resource "vcd_ip_space_ip_allocation" "public-floating-ip-manual" {
+  org_id      = data.vcd_org.org1.id
+  ip_space_id = vcd_ip_space.space1.id
+  type        = "FLOATING_IP"
+  usage_state = "USED_MANUAL"
+  description = "manually used floating IP"
+
+  depends_on = [vcd_nsxt_edgegateway.ip-space]
+}
+
+resource "vcd_nsxt_nat_rule" "dnat-floating-ip" {
+  org             = "cloud"
+  edge_gateway_id = vcd_nsxt_edgegateway.ip-space.id
+
+  name      = "dnat-ip-space-ip"
+  rule_type = "DNAT"
+
+  # Using allocated Floating IP From IP Space
+  external_address = vcd_ip_space_ip_allocation.public-floating-ip.ip_address
+  internal_address = "77.77.77.1"
+  logging          = true
+}
+
+resource "vcd_ip_space_ip_allocation" "public-ip-prefix" {
+  org_id        = data.vcd_org.org1.id
+  ip_space_id   = vcd_ip_space.space1.id
+  type          = "IP_PREFIX"
+  prefix_length = 29
+
+  depends_on = [vcd_nsxt_edgegateway.ip-space]
+}
+
+resource "vcd_network_routed_v2" "using-public-prefix" {
+  org             = "cloud"
+  name            = "ip-space-allocated-prefix"
+  edge_gateway_id = vcd_nsxt_edgegateway.ip-space.id
+
+  # Using prefix allocated from IP Space
+  gateway       = cidrhost(vcd_ip_space_ip_allocation.public-ip-prefix.ip_address, 1)
+  prefix_length = split("/", vcd_ip_space_ip_allocation.public-ip-prefix.ip_address)[1]
+
+  static_ip_pool {
+    start_address = cidrhost(vcd_ip_space_ip_allocation.public-ip-prefix.ip_address, 2)
+    end_address   = cidrhost(vcd_ip_space_ip_allocation.public-ip-prefix.ip_address, 4)
+  }
+}
+```
+
+## Sample configuration without IP Spaces (the original way)
```

Comment:
I would like to see the material rearranged as follows:

1. a few lines of explanation of what we want to achieve
2. A sample of the old usage (this section)
3. A sample of the new usage (the previous section)

The way it is now, we only see a short way of doing things and a long way of doing things, but we can't appreciate where exactly we should apply the changes.
It will be helpful if the samples had commented labels for the corresponding operations in the old and new.

Reply:
Modified